feat(sdk-bundle): stacked and sidebar layouts for metabot questions

heypoom opened the pull request:
Closes EMB-857
Closes EMB-862

Adds the automatic, stacked and sidebar layouts to Embedded Metabot with new designs.

### How to verify

- Start the ai-service.
  - See instruction in https://github.com/metabase/ai-service and see its README docs
  - Make sure to copy the sample env and grab the credentials in 1Password!
- Go to Storybook > MetabotQuestion
- Try out the Full Width, Centered, Stacked and Sidebar stories.

### Demo

Full Width - Auto Layout (sidebar)

<img width="1970" height="1538" alt="CleanShot 2568-09-27 at 21 24 51@2x" src="https://github.com/user-attachments/assets/d631db7c-2167-4bc4-80cf-cfb5adcfd4a1" />

With Padding - Auto Layout (sidebar)

<img width="2256" height="1534" alt="CleanShot 2568-09-27 at 21 25 34@2x" src="https://github.com/user-attachments/assets/28168a68-f736-4f08-ae88-32b9e25340a8" />

With Padding - Auto Layout - Mobile Size (stacked) - scrollbar always visible on OS

<img width="1378" height="1432" alt="image" src="https://github.com/user-attachments/assets/9a1ec4f2-eb92-4f14-9b2b-01ea62c79925" />

### Checklist

- [X] Tests are created (e2e)

heypoom committed:
design updates

heypoom committed:
scroll after new messages arise

heypoom committed:
adjust textarea alignment

heypoom committed:
add centered layout

heypoom committed:
add sidebar and stacked layouts

heypoom committed:
fix sidebar in auto layout

heypoom committed:
expose more flexible size props

heypoom committed:
extract MetabotQuestion types to a separate file

heypoom committed:
add suggestion boxes

heypoom committed:
use 600px breakpoint per design

heypoom committed:
add chatbox textarea overflow menu

heypoom committed:
remove autoLayout from CSS

heypoom committed:
add overflow hover color

heypoom committed:
layout cleanup to css modules

heypoom committed:
refactor embedded metabot components into separate files

heypoom committed:
add e2e test for Metabot suggestions

heypoom committed:
add test for derived layouts

heypoom committed:
refactor derived layout

heypoom committed:
do not memoize suggestedPrompts

heypoom committed:
use css module nesting to simplify styles

heypoom committed:
simplify MetabotQuestion stories

heypoom committed:
expose MetabotQuestion

heypoom committed:
docs: update ai-chat.md with the layout prop

heypoom committed:
update layout prop docstring

heypoom committed:
simplify SidebarInput logic

heypoom committed:
refactor chat input styles and file names

heypoom committed:
refactor MetabotChatSuggestions

heypoom committed:
flatten MetabotQuestion component structure

heypoom committed:
rename content to question

heypoom committed:
document MetabotQuestion

heypoom committed:
Merge branch 'master' into emb-857-sdk-revise-default-layout-for-metabotquestion-component

heypoom committed:
make chatInput transparent

heypoom committed:
remove stack gap for sidebar

heypoom committed:
remove MetabotChatEmbedding

heypoom committed:
allow shift+enter to insert newlines

sanex3339 commented:
On the stacked layout, scroll bars of the question and the chag a bit not aligned, not sure if it's the issue or not

<img width="204" height="1436" alt="image" src="https://github.com/user-attachments/assets/41043b77-7be7-40b4-8674-d6530741a3c2" />

sanex3339 commented on enterprise/frontend/src/embedding-sdk-bundle/components/public/MetabotQuestion/MetabotQuestion.tsx:
```diff
@@ -1,100 +1,102 @@
-
-  if (isLocaleLoading) {
-    return <SdkLoader />;
+  const isSmallScreen = useIsSmallScreen();
```
Maybe better to use content-queries?
I assume it may be embedded inside a container and then better to rely on container dimensions

heypoom replied:
Agree! I was thinking the same but couldn't find a `useContainerQuery` hook (something like [this](https://github.com/joshuaaron/use-container-queries/blob/main/src/index.ts)); I tried to do it via CSS but it needs a lot of styles duplication for the automatic layout and the stacked/sidebar layout, so I'd prefer using JS here. Let me see if I can do something with the element width 🙏🏻

heypoom replied:
Fixed in https://github.com/metabase/metabase/pull/64142/commits/7fe7538220e0efefed7318c6789b30877cec8b01, thanks!

heypoom committed:
export MetabotQuestionProps type

heypoom committed:
make MetabotQuestion always full height

heypoom committed:
better alignment for chat buttons

heypoom committed:
remove default padding of sdk question for Metabot scrollbar

heypoom committed:
remove scrollbar from stories

heypoom committed:
use useElementSize instead of useMedia for size detection

heypoom committed:
remove scrollbar from centered layout

heypoom commented:
> The Sidebar and Stacked layouts have a scroll bar

Fixed in https://github.com/metabase/metabase/pull/64142/commits/bd9789488fbab599ed8cc347522b25e8846c0d6b. This is actually because of how I center the layout in Storybook and wasn't caused by MetabotQuestion.

<img width="1836" height="1530" alt="CleanShot 2568-10-01 at 02 30 12@2x" src="https://github.com/user-attachments/assets/c44dacc5-ffff-4bea-ac0c-1c8603583f69" />

---

> Full-height layout has a non-full-height question

Fixed in https://github.com/metabase/metabase/pull/64142/commits/78cd515d700d819e1e9a18261409f7fba52e258d.

<img width="1382" height="1432" alt="CleanShot 2568-10-01 at 02 31 04@2x" src="https://github.com/user-attachments/assets/247c6e0c-ef2a-4329-90ce-3f89c8f45d8a" />

---

> On the stacked layout, scroll bars of the question and the chat a bit not aligned, not sure if it's the issue or not

Fixed in https://github.com/metabase/metabase/pull/64142/commits/f4f12f998421e15d3b506449936ebbef0339af26, it was caused by padding. Without the padding, now it's all aligned!

<img width="1378" height="1432" alt="CleanShot 2568-10-01 at 02 32 14@2x" src="https://github.com/user-attachments/assets/cea481ee-6e28-47ee-8ce4-91890872c755" />
